internal/controller/monovertexrollout_controller.go

```diff
@@ -0,0 +1,361 @@
+		if err := r.client.Update(ctx, monoVertexRollout); err != nil {
+			r.ErrorHandler(monoVertexRollout, err, "UpdateFailed", "Failed to update MonoVertexRollout")
+			statusUpdateErr := r.updateMonoVertexRolloutStatusToFailed(ctx, monoVertexRollout, err)
+			if statusUpdateErr != nil {
+				r.ErrorHandler(monoVertexRollout, statusUpdateErr, "UpdateStatusFailed", "Failed to update MonoVertexRollout status")
+				return ctrl.Result{}, statusUpdateErr
+			}
+			return ctrl.Result{}, err
+		}
+		// restore original status which is lost during last call to Update()
+		monoVertexRollout.Status = monoVertexRolloutStatus
+	}
+
+	if monoVertexRollout.DeletionTimestamp.IsZero() {
+		statusUpdateErr := r.updateMonoVertexRolloutStatus(ctx, monoVertexRollout)
+		if statusUpdateErr != nil {
+			r.ErrorHandler(monoVertexRollout, statusUpdateErr, "StatusUpdateFailed", "Failed to update MonoVertexRollout")
+			return ctrl.Result{}, statusUpdateErr
+		}
+	}
+
+	r.recorder.Eventf(monoVertexRollout, corev1.EventTypeNormal, "ReconciliationSuccessful", "Reconciliation successful")
+	numaLogger.Debug("reconciliation successful")
+
+	return result, nil
+}
+
+func (r *MonoVertexRolloutReconciler) reconcile(ctx context.Context, monoVertexRollout *apiv1.MonoVertexRollout) (ctrl.Result, error) {
+
+	numaLogger := logger.FromContext(ctx)
+
+	// remove finalizers if monoVertexRollout is being deleted
+	if !monoVertexRollout.DeletionTimestamp.IsZero() {
+		numaLogger.Info("Deleting MonoVertexRollout")
+		if controllerutil.ContainsFinalizer(monoVertexRollout, finalizerName) {
+			controllerutil.RemoveFinalizer(monoVertexRollout, finalizerName)
+		}
+		return ctrl.Result{}, nil
+	}
+
+	if !controllerutil.ContainsFinalizer(monoVertexRollout, finalizerName) {
+		controllerutil.AddFinalizer(monoVertexRollout, finalizerName)
+	}
+
+	newMonoVertexDef := &kubernetes.GenericObject{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "MonoVertex",
+			APIVersion: "numaflow.numaproj.io/v1alpha1",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:            monoVertexRollout.Name,
+			Namespace:       monoVertexRollout.Namespace,
+			OwnerReferences: []metav1.OwnerReference{*metav1.NewControllerRef(monoVertexRollout.GetObjectMeta(), apiv1.MonoVertexRolloutGroupVersionKind)},
+		},
+		Spec: monoVertexRollout.Spec.MonoVertex.Spec,
+	}
+
+	existingMonoVertexDef, err := kubernetes.GetCR(ctx, r.restConfig, newMonoVertexDef, "monovertices")
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			numaLogger.Debugf("MonoVertex %s/%s doesn't exist so creating", monoVertexRollout.Namespace, monoVertexRollout.Name)
+			monoVertexRollout.Status.MarkPending()
+
+			if err := kubernetes.CreateCR(ctx, r.restConfig, newMonoVertexDef, "monovertices"); err != nil {
+				return ctrl.Result{}, err
+			}
+
+			monoVertexRollout.Status.MarkDeployed(monoVertexRollout.Generation)
+		} else {
+			return ctrl.Result{}, fmt.Errorf("error getting MonoVertex: %v", err)
+		}
+	} else {
+		// merge and update
```

Comment:
Can you add a TODO saying for monovertex we can directly apply changes and no need to do draining?

Reply:
Can do, for `apply change directly` what should the logic be?

Reply:
this is what you have right, just apply the spec directly without checking if needs pausing/draining.

Reply:
oh my bad, not TODO, but a comment... sorry for the typo